test/cbmc/sources/core_sntp_cbmc_state.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * coreSNTP v1.0.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_sntp_cbmc_state.c
+ * @brief Implements the functions defined in core_sntp_cbmc_state.h.
+ */
+#include <stdint.h>
+#include <stdlib.h>
+#include "core_sntp_client.h"
+#include "core_sntp_cbmc_state.h"
+#include "core_sntp_stubs.h"
+
+SntpContext_t * allocateCoreSntpContext( SntpContext_t * pContext )
```

Comment:
Why does this function take in a `SntpContext_t *` parameter as well as have the same return type? 
Should it be fine to just have no parameter for this function?

Reply:
I have used it as input parameter because it give flexibility to allocate pContext both in harness as well as in the stub if input pContext is NULL.

Reply:
What is the value of having that flexibility? As all the members of the context are anyway allocated within the function, it would just be cleaner to have the memory of `SntpContext_t` be also completely handled by the function..